java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -275,16 +281,23 @@ class DatabaseImpl(
         db: SQLiteDatabase
     ): Any = db.rawQuery(
         """
-            SELECT entity_id, backing_storage_key, version_map
+            SELECT
+                entity_id,
+                creation_timestamp,
+                expiration_timestamp,
+                backing_storage_key,
+                version_map
             FROM entity_refs
             WHERE id = ?
         """.trimIndent(),
         arrayOf(entityRefId.toString())
     ).forSingleResult {
         Reference(
             id = it.getString(0),
-            storageKey = StorageKeyParser.parse(it.getString(1)),
-            version = it.getVersionMap(2)
+            storageKey = StorageKeyParser.parse(it.getString(3)),
+            version = it.getVersionMap(4),
+            creationTimestamp = it.getLong(1),
+            expirationTimestamp = it.getLong(2)
```

Comment:
I'd suggest reordering these in order of `it.getLong(i)` (or reordering the SELECT query), just to make it a bit more obvious.

Reply:
done